packages/x-date-pickers/src/AdapterLuxon/AdapterLuxon.ts

```diff
@@ -353,7 +353,11 @@ export class AdapterLuxon implements MuiPickersAdapter<DateTime, string> {
   };
 
   public startOfWeek = (value: DateTime) => {
-    return value.startOf('week');
+    return value.startOf(
+      'week',
+      // @ts-ignore
+      { useLocaleWeeks: true },
```

Comment:
The types do not include the new APIs yet, I can remove those comment later.
For older versions of Luxon, this param should be ignored and the behavior should just be like before. 
I think we can bump the minimal version of Luxon in v8 to unify everyones behavior, but keep supporting older one in v7 to avoid forcing a very-recent version.